trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/chandler/IMemoryAdressing.java

```diff
@@ -0,0 +1,251 @@
+package de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.base.chandler;
+
+import java.math.BigInteger;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+
+import de.uni_freiburg.informatik.ultimate.boogie.ast.AssumeStatement;
+import de.uni_freiburg.informatik.ultimate.boogie.ast.Declaration;
+import de.uni_freiburg.informatik.ultimate.boogie.ast.Expression;
+import de.uni_freiburg.informatik.ultimate.boogie.ast.IdentifierExpression;
+import de.uni_freiburg.informatik.ultimate.boogie.ast.Specification;
+import de.uni_freiburg.informatik.ultimate.boogie.ast.Statement;
+import de.uni_freiburg.informatik.ultimate.boogie.ast.VariableLHS;
+import de.uni_freiburg.informatik.ultimate.boogie.type.BoogieType;
+import de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.base.chandler.TypeSizeAndOffsetComputer.Offset;
+import de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.container.c.CPointer;
+import de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.container.c.CPrimitive;
+import de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.container.c.ICType;
+import de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.result.ExpressionResult;
+import de.uni_freiburg.informatik.ultimate.cdt.translation.implementation.result.RValue;
+import de.uni_freiburg.informatik.ultimate.core.model.models.ILocation;
+import de.uni_freiburg.informatik.ultimate.plugins.generator.cacsl2boogietranslator.preferences.CACSLPreferenceInitializer.CheckMode;
+import de.uni_freiburg.informatik.ultimate.util.datastructures.relation.Pair;
+import de.uni_freiburg.informatik.ultimate.util.datastructures.relation.Triple;
+
+/**
+ * The interface defining the functions for the different addressing modes.
+ */
+public interface IMemoryAdressing {
+	/**
+	 * Constructs the required metadata for the selected addressing mode.
+	 *
+	 * @param requiredFeatures
+	 *            The Features that are currently needed for the program to be verified.
+	 * @return The metadata declarations.
+	 */
+	List<Declaration> constructMetaData(RequiredMemoryModelFeatures requiredFeatures);
+
+	/**
+	 * Returns a list of metadata declarations needed for the memory model infrastructure.
+	 *
+	 * @return The declarations.
+	 */
+	List<MemoryModelDeclarations> metaDataDeclarations();
```

Comment:
Add prefix `get` to interface method name.

```suggestion
	List<MemoryModelDeclarations> getMetaDataDeclarations();
```

Reply:
See [bf2a925](https://github.com/ultimate-pa/ultimate/commit/bf2a9251bf23c01929004e0a22a407d1c43a4f33)